Review getting started (#1895)
webpack/webpack.js.org: src/content/guides/getting-started.md

  - TheDutchCoder
  - sudarsangp
  - Vanguard90
  - chrischinchilla
---

Webpack is used to compile JavaScript modules. Once [installed](/guides/installation), you can interface with webpack either from its [CLI](/api/cli) or [API](/api/node). If you're still new to webpack, read through the [core concepts](/concepts) and [this comparison](/comparison) to learn why to use it over the other tools available the community.

## Basic Setup

Create a directory, initialize npm, and [install webpack locally](/guides/installation#local-installation):

```bash
mkdir webpack-demo && cd webpack-demo
npm init -y
npm install --save-dev webpack
```

Create the following directory structure and contents:

> **project**

Reviewer: We use double underscores everywhere. It makes no difference, but no need to change it.

Author: Sorry, that's my Markdown linter there, always changes the formatting used :(

</html>
```

In this example, there are implicit dependencies between the `<script>` tags. The `index.js` file depends on including `lodash` in the page before it runs. This is because `index.js` never declared a need for `lodash`; it assumes that the global variable `_` exists.

There are problems with managing JavaScript projects this way:

> -   It is not immediately apparent that the script depends on an external library.

Reviewer: No need for double spaces here?

Author: Again it's my markdown linter, apparently it's 'more correct', but somewhat irrelevant.

Reviewer 2: This can bring more fragmentation to the content, so I would suggest reversing these type of changes since they don't add value and make the PR even bigger and longer to review :)

## Modules

[ES2015](https://babeljs.io/learn-es2015/) standardized the [`import`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) and [`export`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export) statements and [most browsers support them](https://caniuse.com/#search=modules). Some older browsers still lag behind but webpack supports modules out of the box.

Behind the scenes, webpack "transpiles" the code so that older browsers can also run it. If you inspect `dist/main.js`, you might be able to see how webpack does this, it's quite ingenious! Besides `import` and `export`, webpack supports other module syntaxes as well, see [Module API](/api/module-methods) for more information.

Webpack does not alter any code other than `import` and `export` statements. If you are using other [ES2015 features](http://es6-features.org/), make sure to [use a transpiler](/loaders/#transpiling) such as [Babel](https://babeljs.io/) or [Bublé](https://buble.surge.sh/guide/) via webpack's [loader system](/concepts/loaders/).

> Webpack doesn't alter any code other than `import` and `export` statements. If you are using other [ES2015 features](http://es6-features.org/), make sure to [use a transpiler](/loaders/#transpiling) such as [Babel](https://babeljs.io/) or [Bublé](https://buble.surge.sh/guide/) via webpack's [loader system](/concepts/loaders/).

Author: ```suggestion

```

    + 1 hidden module
```

W> Note that when calling `webpack` via its path on Windows, you must use backslashes instead, e.g. `node_modules\.bin\webpack --config webpack.config.js`.

> T> If a `webpack.config.js` is present, the `webpack` command picks it up by default. We use the `--config` option here to show that you can pass a config of any name. This is be useful for more complex configurations that need to be split into multiple files.

Reviewer: g/is be/is

    + 1 hidden module
```

> T> We can pass custom parameters to webpack by adding two dashes between the `npm run build` command and your parameters, e.g. `npm run build -- --colors`.

Reviewer: g/your/our

## Conclusion

> Now that we have a basic build you should move on to the next guide [`Asset Management`](/guides/asset-management) to learn how to manage assets like images and fonts with webpack. At this point, your project should look like this:

Reviewer: g/you/we
g/your/our

Reviewer 2: I agree with @TheDutchCoder that is better to speak to the reader in _you_ and not as _we_

Author: I agree too, but that's what was here before, I was making it consistent across the document. Consistency is the most important, wether it's we/you doesn't matter as much.

You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/concepts/mode/
```

> T> If a `webpack.config.js` is present, the `webpack` command picks it up by default. We use the `--config` option here to show that you can pass a config of any name. This is useful for more complex configurations that you need to split into multiple files.

Author: ```suggestion
T> If a `webpack.config.js` file is present in the root directory, the `webpack` command picks it up by default. We use the `--config` option here to demonstrate how to tell webpack to use a specific configuration file.
```

  }
```

> Now we can use the `npm run build` command in place of the `npx` command we used earlier. Within `scripts` we can reference locally installed npm packages by name the same way we did with `npx`. This convention is the standard in most npm-based projects because it allows all contributors to use the same set of common scripts (each with flags like `--config` if necessary).

Reviewer 3: ```suggestion
Now we can use the `npm run build` command in place of the `npx` command we used earlier. Within `scripts` we can reference locally installed npm packages by their names as we did with `npx`. This convention is the standard in most npm-based projects because it allows all contributors to use the same set of common scripts (each with flags like `--config` if necessary).
```

You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/concepts/mode/.
```

T> We can pass custom parameters to webpack by adding two dashes between the `npm run build` command and the parameters, e.g. `npm run build -- --colors`.

## Conclusion

> With a basic build complete, we recommend you read the next guide [`Asset Management`](/guides/asset-management) to learn how to manage assets like images and fonts with webpack. At this point, the project should look like this:

Reviewer 3: ```suggestion
With a basic build complete, we recommend you read the [`Asset Management guide`](/guides/asset-management/) to learn how to manage assets like images and fonts with webpack. At this point, the project should look like this:
```

Author: How do you know this guide is next?

T> If you're using npm 5, you'll probably also see a `package-lock.json` file in your directory.

> If you want to learn more about webpack's design, you can check out the [basic concepts](/concepts) and [configuration](/configuration) pages. Furthermore, the [API](/api) section digs into the interfaces webpack offers.

Reviewer 3: Trailing slashes for links to also work on dev server
```suggestion
If you want to learn more about webpack's design, you can check out the [concepts](/concepts/) and [configuration](/configuration/) pages. Furthermore, the [API](/api/) section digs into the interfaces webpack offers.
```
